### ISSUE TITLE
xSQLServerSetup: ADVANCEDANALYTICS feature needs its own licence acceptance switch

### ISSUE THREAD

**Issue by sjethapason** [PowerShell/xSQLServer#340]
**Details of the scenario you try and problem that is occurring:**
xSQLServer installation with ADVANCEDANALYTICS in the features string

**The DSC configuration that is using the resource (as detailed as possible):**
```
foreach($SQLServer in $Node.SQLServers)
{
        $SQLInstanceName = $SQLServer.InstanceName

        $Features = "SQLENGINE,ADVANCEDANALYTICS,FULLTEXT,DQ,RS,SQL_SHARED_MR,DQC,CONN,IS,BOL,SNAC_SDK,MDS"

        if($Features -ne "")
        {
            xSqlServerSetup ($Node.NodeName + $SQLInstanceName)
                {
                    # DependsOn = "[WindowsFeature]NET-Framework-Core"
                    SourcePath = $Node.SourcePath
                    SourceFolder = $Node.SourceFolder
                    SetupCredential = $Node.InstallerServiceAccount
                    InstanceName = $SQLInstanceName
                    Features = $Features
                    SQLSysAdminAccounts = $Node.InsallerServiceAccount
                    InstallSharedDir = "C:\Program Files\Microsoft SQL Server"
                    InstallSharedWOWDir = "C:\Program Files (x86)\Microsoft SQL Server"
                    InstanceDir = "C:\Program Files\Microsoft SQL Server"
                    InstallSQLDataDir = "C:\Program Files\Microsoft SQL Server\MSSQL11.MSSQLSERVER\MSSQL\Data"
                    SQLUserDBDir = "C:\Program Files\Microsoft SQL Server\MSSQL11.MSSQLSERVER\MSSQL\Data"
                    SQLUserDBLogDir = "C:\Program Files\Microsoft SQL Server\MSSQL11.MSSQLSERVER\MSSQL\Data"
                    SQLTempDBDir = "C:\Program Files\Microsoft SQL Server\MSSQL11.MSSQLSERVER\MSSQL\Data"
                    SQLTempDBLogDir = "C:\Program Files\Microsoft SQL Server\MSSQL11.MSSQLSERVER\MSSQL\Data"
                    SQLBackupDir = "C:\Program Files\Microsoft SQL Server\MSSQL11.MSSQLSERVER\MSSQL\Data"
                    ASDataDir = "C:\Program Files\Microsoft SQL Server\MSAS11.MSSQLSERVER\OLAP\Data"
                    ASLogDir = "C:\Program Files\Microsoft SQL Server\MSAS11.MSSQLSERVER\OLAP\Log"
                    ASBackupDir = "C:\Program Files\Microsoft SQL Server\MSAS11.MSSQLSERVER\OLAP\Backup"
                    ASTempDir = "C:\Program Files\Microsoft SQL Server\MSAS11.MSSQLSERVER\OLAP\Temp"
                    ASConfigDir = "C:\Program Files\Microsoft SQL Server\MSAS11.MSSQLSERVER\OLAP\Config"
                }
```
**Version of the Operating System, SQL Server and PowerShell the DSC Target Node is running:**
OS: Windows Server 2012 R2
SQL Server: 2016 Developer Edition
DSC Target Node: 5.0.10586.117

**Version of the DSC module you're using, or 'dev' if you're using current dev branch:**
4.0.0.0

**Comment by johlju** [comment]
Looks like the switch is /IACCEPTREVOLICENSETERMS. This should be added if ADVANCEDANALYTICS is a feature.
https://msdn.microsoft.com/en-us/library/mt590808.aspx

I add this as a bug. Feel more like a bug than an enhancement. :)

**Comment by kewalaka** [comment]
@johlju I think it should be [/IACCEPTROPENLICENSETERMS ](https://docs.microsoft.com/en-us/sql/advanced-analytics/r/unattended-installs-of-sql-server-r-services), it would be useful to add support for /MRCACHEDIRECTORY at the same time, to allow an install of the R Services without internet access.  Plainly, requires a previous process to put the R files in that location.  There's also Python support [just added](https://docs.microsoft.com/en-us/sql/advanced-analytics/python/unattended-installs-of-sql-server-python-services) too, though perhaps that should be a different issue.

**Comment by johlju** [comment]
Yes you are right. I don't know what I copied in. Thanks for correcting me! 😄 

I opened new issue for argument MRCACHEDIRECTORY in issue #588.
And pointed to the article for Python install in issue #555.

**Comment by johlju** [comment]
@marcinbojko yes none in the community has picked this one up and run with it so the issue still there. 

I labeled this 'good first issue' since it's an easier addition to the resource (code change, adding tests and documentation).

**Comment by marcinbojko** [comment]
@johlju Thanks, I've noticed. Just had some hopes though ;)

**Comment by johlju** [comment]
@marcinbojko I understand. 🙂 You are always welcome to give this a try your self, are you up to contribute here? 😃 I happy to help you get started, or if you get stuck.

**Comment by marcinbojko** [comment]
@johlju Happy too, but I think my skillset is too low ;)

**Comment by johlju** [comment]
@marcinbojko if you like PowerShell and eager to learn your skill set is not to low. :) 

To get started read [Getting Started with GitHub](https://github.com/PowerShell/DscResources/blob/master/GettingStartedWithGitHub.md) to set up Git. Suggest you install Visual Studio Code with the PowerShell extenstion installed. After you forked (button top right on this page) then just do this.

```powershell
mkdir c:\source
cd c:\source
git clone https://github.com/PowerShell/SqlServerDsc
cd c:\source\SqlServerDsc
git remote add my https://github.com/marcinbojko/SqlServerDsc
git checkout -b working-branch
code .
```

Now you have a clean working branch named 'working-branch' and ready to start changing and committing changes to your local working branch.
Push your working branch to your fork by running.

```
git push my working-branch
```

When you feel your finished. Got into your fork https://github.com/marcinbojko/SqlServerDsc and choose you working branch and click 'new pull request'. 

Happy coding! 😃  Happy to help if you get stuck.

**Comment by nesith** [comment]
I am looking into fixing this issue, I was thinking that while fixing this, I could also look at adding support for using Parent feature parameter
